denonavr/api.py

```diff
@@ -309,3 +313,184 @@ def prepare_appcommand_body(cmd_list: Tuple[AppCommandCmd]) -> bytes:
     def is_default_async_client(self) -> bool:
         """Check if default httpx.AsyncCLient getter is used."""
         return self.async_client_getter is get_default_async_client
+
+@attr.s(auto_attribs=True, hash=False, on_setattr=DENON_ATTR_SETATTR)
+class DenonAVRTelnetApi:
+
+    host: str = attr.ib(converter=str, default="localhost")
+    timeout: int = attr.ib(converter=int, default=10)
+    _telnet_task: asyncio.Task = attr.ib(default=None)
+    _callbacks: dict[str, Callable] = attr.ib(
+        validator=attr.validators.instance_of(dict),
+        default={}, init=False
+    )
+    
+    async def async_connect(self) -> asyncio.Task:
+        """ Connect to the receiver asynchronously."""
+        try:
+            self._socket_reader, self._socket_writer = await asyncio.wait_for(asyncio.open_connection(self.host, 23), timeout=self.timeout)
+            self._telnet_task = asyncio.create_task(self._async_monitor())
+            return self._telnet_task
+        except asyncio.exceptions.TimeoutError as err:
+            _LOGGER.debug(
+                "Socket timeout exception on connect",
+                exc_info=True)
+            raise AvrTimoutError(
+                "TimeoutException: {}".format(err), "connect") from err
```

Comment:
```suggestion
        """ Connect to the receiver asynchronously."""
        try:
            self._socket_reader, self._socket_writer = await asyncio.wait_for(asyncio.open_connection(self.host, 23), timeout=self.timeout)
        except asyncio.exceptions.TimeoutError as err:
            _LOGGER.debug(
                "Socket timeout exception on connect",
                exc_info=True)
            raise AvrTimoutError(
                "TimeoutException: {}".format(err), "connect") from err
        self._telnet_task = asyncio.create_task(self._async_monitor())
        return self._telnet_task
```
It looks cleaner if you only put the statements which could raise the exception between `try` and `except`.